homeassistant/components/volumio/browse_media.py

```diff
@@ -0,0 +1,157 @@
+"""Support for media browsing."""
+import json
+
+from homeassistant.components.media_player import BrowseError, BrowseMedia
+from homeassistant.components.media_player.const import (
+    MEDIA_CLASS_ALBUM,
+    MEDIA_CLASS_ARTIST,
+    MEDIA_CLASS_CHANNEL,
+    MEDIA_CLASS_DIRECTORY,
+    MEDIA_CLASS_GENRE,
+    MEDIA_CLASS_PLAYLIST,
+    MEDIA_CLASS_TRACK,
+    MEDIA_TYPE_MUSIC,
+)
+
+PLAYABLE_ITEM_TYPES = [
+    "folder",
+    "song",
+    "mywebradio",
+    "webradio",
+    "playlist",
+    "cuesong",
+    "remdisk",
+    "cuefile",
+    "folder-with-favourites",
+    "internal-folder",
+]
+
+NON_EXPANDABLE_ITEM_TYPES = [
+    "song",
+    "webradio",
+    "mywebradio",
+    "cuesong",
+    "album",
+    "artist",
+    "cd",
+    "play-playlist",
+]
+
+
+def _item_to_children_media_class(item, info=None):
+    if info and "album" in info and "artist" in info:
+        return MEDIA_CLASS_TRACK
+    if item["uri"].startswith("playlists"):
+        return MEDIA_CLASS_PLAYLIST
+    if item["uri"].startswith("artists://"):
+        if len(item["uri"]) > 10:
+            return MEDIA_CLASS_ALBUM
+        return MEDIA_CLASS_ARTIST
+    if item["uri"].startswith("albums://"):
+        if len(item["uri"]) > 9:
+            return MEDIA_CLASS_TRACK
+        return MEDIA_CLASS_ALBUM
+    if item["uri"].startswith("genres://"):
+        if len(item["uri"]) > 9:
```

Comment:
```suggestion
    if item["uri"].startswith(GENRES_URL_PREFIX):
        if len(item["uri"]) > len(GENRES_URL_PREFIX):
```

Reply:
Another way would be to lstrip the prefix from the item uri and check if we have something left.